### PR TITLE
GCP IPI cluster with kms enabled

### PR DESCRIPTION
**Pull request by jianli-wei**
1. The QE sub-task: [OCPQE-13126](https://issues.redhat.com/browse/OCPQE-13126) Base Profile - GCP IPI cluster with kms enabled
2. add the workflow `cucushift-installer-rehearse-gcp-ipi-disk-encryption` and its provision chain
3. add step `ipi-conf-gcp-osdisk-encryption-key` for inserting OS disk encryption configurations into install-config.yaml
4. add step `cucushift-installer-check-gcp-disk-encryption` which validates the OS disk custom encryptions settings on cluster nodes

### CODE REVIEW

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/osdisk-custom-encryption/provision/cucushift-installer-rehearse-gcp-ipi-osdisk-custom-encryption-provision-chain.metadata.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+	"path": "cucushift/installer/rehearse/gcp/ipi/osdisk-custom-encryption/provision/cucushift-installer-rehearse-gcp-ipi-osdisk-custom-encryption-provision-chain.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/osdisk-custom-encryption/provision/cucushift-installer-rehearse-gcp-ipi-osdisk-custom-encryption-provision-chain.metadata.json:2]: How about `gcp/ipi/disk/encryption/`, we might extend other disk feature integration.

File: ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [[ "${CONTROL_PLANE_OS_DISK_CUSTOM_ENCRYPTION}" != "yes" ]] && [[ "${COMPUTE_OS_DISK_CUSTOM_ENCRYPTION}" != "yes" ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh:7]: When calling the step in some chain/workflow, that means users want to test the feature on purpose, so we do not need the two ENV vars and the condition.

File: ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  platform:
+    gcp:
+      osDisk:
+        encryptionKey:
+          kmsKey:
+            keyRing: ${KMS_KEY_RING}
+            location: ${KMS_KEY_RING_LOCATION}
+            name: ${KMS_KEY_NAME}
+            projectID: ${project_id}
+          kmsKeyServiceAccount: ${service_account}
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh:57]: `controlPlane` does not need the `kmsKeyServiceAccount` setting?

File: ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+fi
+service_account=$(jq -r .client_email ${CLUSTER_PROFILE_DIR}/gce.json)
+if [[ "${KMS_KEY_SERVICE_ACCOUNT}" != "" ]]; then
+  service_account="${KMS_KEY_SERVICE_ACCOUNT}"
+fi
+
+if [[ "${CONTROL_PLANE_OS_DISK_CUSTOM_ENCRYPTION}" == "yes" ]]; then
+  cat > "${PATCH}" << EOF
+controlPlane:
+  name: master
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh:30]: I think we do not need this `name` setting.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/osdisk-custom-encryption/provision/cucushift-installer-rehearse-gcp-ipi-osdisk-custom-encryption-provision-chain.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+chain:
+  as: cucushift-installer-rehearse-gcp-ipi-osdisk-custom-encryption-provision
+  steps: 
+    - chain: ipi-conf-gcp
+    - ref: ipi-conf-gcp-osdisk-encryption-key
+    - chain: ipi-install
+    - ref: gcp-provision-check-osdisk-encryption
+    - ref: enable-qe-catalogsource
+  env:
+  - name: CONTROL_PLANE_OS_DISK_CUSTOM_ENCRYPTION
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/osdisk-custom-encryption/provision/cucushift-installer-rehearse-gcp-ipi-osdisk-custom-encryption-provision-chain.yaml:10]: How about rename it to `CONTROL_PLANE_DISK_ENCRYPTION` for a shorter name.

File: ci-operator/step-registry/gcp/provision/check-osdisk-encryption/gcp-provision-check-osdisk-encryption-ref.metadata.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+	"path": "gcp/provision/check-osdisk-encryption/gcp-provision-check-osdisk-encryption-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/gcp/provision/check-osdisk-encryption/gcp-provision-check-osdisk-encryption-ref.metadata.json:2]: How about moving the folder to `ci-operator/step-registry/cucushift/installer/check/gcp/disk-encryption` for the post check.

File: ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh (hunk 3 above)
Comment (jianli-wei) [reply ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh:57]: Per [the QE test case](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-36887) and [flexy templates](https://gitlab.cee.redhat.com/aosqe/flexy-templates/-/blob/master/functionality-testing/aos-4_13/hosts/launch_opts/common_ipi_env_v1.rb#L936-941), it's said "Master can only be processed with the default compute service account".

File: ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+            projectID: ${project_id}
+EOF
+  yq-go m -x -i "${CONFIG}" "${PATCH}"
+  echo "Updated controlPlane.platform.gcp.osDisk.encryptionKey in '${CONFIG}'."
+fi
+
+if [[ "${COMPUTE_DISK_ENCRYPTION}" == "yes" ]]; then
+  cat > "${PATCH}" << EOF
+compute:
+  platform:
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/gcp/osdisk/encryption-key/ipi-conf-gcp-osdisk-encryption-key-commands.sh:47]: Missing a leading `-` here.

**Pull request merged**